agent/src/main/java/com/teamscale/jacoco/agent/upload/HttpZipUploaderBase.java

```diff
@@ -118,31 +134,46 @@ protected boolean tryUpload(CoverageFile coverageFile) {
 	}
 
 	/**
-	 * Creates the zip file in the system temp directory to upload which includes the given coverage XML and all {@link
-	 * #additionalMetaDataFiles}. The file is marked to be deleted on exit.
+	 * Creates the zip file in the system temp directory to upload which includes
+	 * the given coverage XML and all {@link #additionalMetaDataFiles}. The file is
+	 * marked to be deleted on exit.
 	 */
 	private File createZipFile(CoverageFile coverageFile) throws IOException {
 		File zipFile = Files.createTempFile(coverageFile.getNameWithoutExtension(), ".zip").toFile();
 		zipFile.deleteOnExit();
 		try (FileOutputStream fileOutputStream = new FileOutputStream(zipFile);
-			 ZipOutputStream zipOutputStream = new ZipOutputStream(fileOutputStream)) {
+				ZipOutputStream zipOutputStream = new ZipOutputStream(fileOutputStream)) {
 			fillZipFile(zipOutputStream, coverageFile);
+			fileOutputStream.close();
```

Comment:
Isn't that handled by the try with resources block?

Reply:
Apparently it is. I added some close statements, as this lead to failing tests, I guess only the one in the copy method was actually needed.